<!doctype html>
<html>
	<head>
		<meta charset="utf-8">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

		<title>DE Genes Network Analysis</title>

		<link rel="stylesheet" href="../../css/reveal.css">
		<link rel="stylesheet" href="../../css/theme/cytoscape-protocol-1.css">

		<!-- Theme used for syntax highlighting of code -->
		<link rel="stylesheet" href="../../lib/css/zenburn.css">

		<!-- Printing and PDF exports -->
		<script>
			var link = document.createElement( 'link' );
			link.rel = 'stylesheet';
			link.type = 'text/css';
			link.href = window.location.search.match( /print-pdf/gi ) ? '../../css/print/pdf.css' : '../../css/print/paper.css';
			document.getElementsByTagName( 'head' )[0].appendChild( link );
		</script>
	</head>
	<body>
		<div class="reveal">
			<div class="slides">
<!-- ========================================================== -->
<!-- THIS IS WHERE YOUR CONTENT BEGINS -->

<section>
    <h1>Differentially Expressed Genes Network Analysis</h1>
    <p><highlight>Cytoscape is an open source software platform for integrating, visualizing, and analyzing measurement data in the context of networks.</highlight></p>
    <p>This protocol describes a network analysis workflow in Cytoscape for a set of differentially expressed genes. Points covered:</p>
    <ul>
        <li>Retrieving relevant networks from public databases</li>
        <li>Network functional enrichment analysis</li>
        <li>Integration and visualization of experimental data</li>
        <li>Exporting network visualizations</li>
    </ul>
    <div class="footer" style="position: absolute; left: -3em; bottom: -4em; ">
<img class="plain" src="../../assets/images/Cytoscape.png" style="height:70px;opacity:0.5"> 
<version>3.8</version></div>
</section>

<section>
    <h1>Setup</h1>
    <ul>
    <li>If you haven't already, install the <a href="http://apps.cytoscape.org/apps/stringapp" target="_blank">STRINGapp</a> from the Cytoscape App Store, or install from Cytoscape via <highlight>Apps &rarr; App Manager ...</highlight></li>
    </ul>
</section>

<section>
<h1>Background</h1>
    <p>Ovarian serous cystadenocarcinoma is a type of epithelial ovarian cancer which accounts for ~90% of all ovarian cancers. The data used in this protocol are from <a href="https://cancergenome.nih.gov/" target="_blank">The Cancer Genome Atlas</a>, in which multiple subtypes of serous cystadenocarcinoma were identified and characterized by mRNA expression.</p> <p>We will focus on the differential gene expression between two subtypes, <b>Mesenchymal</b> and <b>Immunoreactive</b>.</p>
    <p>For convenience, the data have already been analyzed and pre-filtered, using log fold change value and adjusted p-value.</p>
</section>
                
<section>
    <h1>Network Retrieval</h1>
    <p>Many public databases and multiple Cytoscape apps allow you to retrieve a network or pathway relevant to your data. For this workflow, we will use the STRING app. Some other options include:</p>
    <ul>
    <li><a href="https://cytoscape.org/cytoscape-tutorials/protocols/wikipathways-app/" target="_blank">WikiPathways</a></li>
    <li><a href="http://www.ndexbio.org/" target="_blank">NDEx</a></li>
    <li><a href="https://genemania.org/" target="_blank">GeneMANIA </a></li>
    </ul>
</section>
                
<section>
    <h1>Retrieve Networks from STRING</h1>
    <p>To identify a relevant network, we will query the STRING database in two different ways:</p>
    <ul>
        <li>Query <b>STRING protein</b> with the list of differentially expressed genes.</li>
        <li>Query <b>STRING disease</b> for ovarian cancer.</li>
    </ul>                
</section>
                
<section>
<h4>STRING Protein Query: Up-regulated genes</h4>
<ul>
    <li>Open the file containing the list of up-regulated genes, <a href="https://cytoscape.github.io/cytoscape-tutorials/protocols/data/TCGA-Ovarian-MesenvsImmuno_UP.txt" target="_blank">TCGA-Ovarian-MesenvsImmuno_UP.txt</a>, and copy the contents to the clipboard.</li>
    <li>In the <highlight>Network Search</highlight> bar at the top of the <highlight>Network Panel</highlight>, select <highlight>STRING protein query</highlight> from the drop-down, and paste in the list of up-regulated genes.</li>
   <li>Open the options panel <img class="plain" src="string-options.png" style="height:30px"> and confirm you are searching <b>Homo sapiens</b> with a cutoff of <b>0.4</b> and <b>0</b> maximum additional interactors.</li>
    <li>Click the search icon <img class="plain" src="SearchIcon.png" style="height:30px"> to search. The resulting network will load automatically.</li>
</ul>
</section>

<section>
<h4>STRING Protein Query: Up-regulated genes</h4>
    <p>The resulting network contains up-regulated genes recognized by STRING, and interactions between them with an evidence score of 0.4 or greater.</p>
<img class="plain" src="string-up.png" style="height:700px"> 
</section>
                
<section>
   <h1>Enrichment Analysis Options</h1> 
    <p>Next, we are going to perform enrichment anlaysis uing the STRING app. Note that there are several other options, including:</p>
    <ul>
        <li><a href="http://bioconductor.org/packages/release/bioc/vignettes/clusterProfiler/inst/doc/clusterProfiler.html" target="_blank">clusterProfiler</a>: An R package for ORA and GSEA</li>
        <li><a href="https://biit.cs.ut.ee/gprofiler/" target="_blank">g-Profiler</a>: An enrichment analysis website</li>
	<li><a href="http://amp.pharm.mssm.edu/Enrichr/" target="_blank">EnrichR</a>: A website that performs enrichment against dozens of ontologies and pathway resources</li>
        <li><a href="http://apps.cytoscape.org/apps/cluego" target="_blank">ClueGO</a>: Creates and visualizes a functionally grouped network of terms/pathways</li>
        <li><a href="http://apps.cytoscape.org/apps/bingo" target="_blank">BiNGO</a>: GO overrepresentation analysis on networks</li>
        <li><a href="https://cytoscape.org/cytoscape-tutorials/protocols/enrichmentmap-pipeline/" target="_blank">EnrichmentMap</a>: Visualize the results of gene-set enrichment as a network</li>
    </ul>
</section>
                
                
<section>
<h1>STRING Enrichment</h1>
    <p>The STRING app has built-in enrichment analysis functionality, which includes enrichment for GO Process, GO Component, GO Function, InterPro, KEGG Pathways, and PFAM.</p>
    <ul>
    <li>With the STRING results network selected, go to <highlight>Apps &rarr; STRING Enrichment &rarr; Retrieve functional enrichment</highlight>.</li>
    <li>When the enrichment analysis is complete, a new tab titled <highlight>STRING Enrichment</highlight> will open in the <highlight>Table Panel</highlight>.</li>
    </ul><br /><br />
     <div class="fragment">
<img class="plain" src="string-enrichment.png">    
</div>
</section>
                
<section>
    <h1>STRING Enrichment</h1>
    <p>The STRING app includes several options for filtering and displaying the enrichment results. The features are all available at the top of the <highlight>STRING Enrichment</highlight> tab.</p>    
    <ul>
        <li>We are going to filter the table to only show GO Process. At the top left of the STRING enrichment tab, click the filter icon <img class="plain" src="FilterIcon.png" style="height:30px">. Select <highlight>GO Process</highlight> and check the <highlight>Remove redundant terms</highlight> check-box. Then click OK.</li>
        <li>Next, we will add a split donut chart to the nodes representing the top terms by clicking on <img class="plain" src="ChartIcon.png" style="height:30px">.</li>
	<li>Explore custom setting via <img class="plain" src="SettingsIcon.png" style="height:30px"> in the top right of the STRING enrichment tab.</li>
    </ul>
    <br /><br />
    <div class="fragment">
<img class="plain" src="string-enrichment-chart.png" style="height:400px">    
</div>    
</section>
                                               
<section>
    <h1>STRING Protein Query: Down-regulated genes</h1>
    <p>Repeat the network search, enrichment analysis and visualization for the set of down-regulated genes:</p>
    <ul>
    <li>Query STRING protein with the list of symbols in <a href="https://cytoscape.github.io/cytoscape-tutorials/protocols/data/TCGA-Ovarian-MesenvsImmuno_DOWN.txt" target="_blank">TCGA-Ovarian-MesenvsImmuno_DOWN.txt</a>.</li>
    <li>Perform STRING Enrichment analysis on the resulting network.</li>
    <li>Filter the analysis results for non-redundant GO Process terms only, add split donut charts. To distinguish between the visualizations of up- and down-regulated results, pick a different color palette under <highlight>Change Color Palette</highlight> under <highlight>Settings</highlight>.</li>
    </ul>
    <div class="fragment">
	    <p><i>Pro-tip: If you remove the Fill Color mapping from the Style Panel (right-click > Edit > Remove...), set the default to light gray, change the split donut to a Pie Chart in <highlight>Settings</highlight> and then try out some of the layouts (see Layout menu), you can end up with network views like this...</i></p>
 <img class="plain" src="string-de-down.png" style="height:370px">    
</div>
</section>

<section>
    <h1>STRING Disease Query</h1>
    <p>Now, we will query the <b>STRING disease</b> database to retrieve a network of ovarian cancer associated genes, completely independent of our dataset.</p>
    <ul>
    <li>In the <highlight>Network Search</highlight> bar at the top of the <highlight>Network Panel</highlight>, select <highlight>STRING disease query</highlight> from the drop-down, and type in <b>ovarian cancer</b>.</li>
    <li>Click enter, or the search icon, to search.</li>
    </ul>
    <div class="fragment">
<p>This will bring in the top 100 ovarian cancer associated genes connected with a confidence score greater than 0.4. Note that because we removed the default STRING node fill color in the previous step, we have lost the "glass ball" effect. For this protocol we don't need it, but if you want to add back that effect, select <highlight>Apps &rarr; STRING &rarr; Enable STRING glass balls effect</highlight>.</p>
<img class="plain" src="string-ovarian.png" style="height:445px">    
</div>
</section>  
                
<section>
    <h1>Data integration</h1>
    <p>Next we will import log fold changes and p-values from our TCGA dataset and use them to create a visualization. Since the network and data use different identifiers, we first have to do some quick identifier mapping. In this case, we will use the gene symbol in the <b>display name</b> column to retrieve <b>Entrez Gene</b> identifiers.</p>
    <ul>
        <li>In the <highlight>Table Panel</highlight>, right-click on the column header for <b>display name</b> and select <highlight>Map column...</highlight>. In the <highlight>ID Mapping</highlight> interface, select <b>Human</b> as species, <b>HGNC</b> as <highlight>Map from</highlight>, and <b>Entrez</b> as <highlight>To</highlight>. Click <highlight>OK</highlight> to continue.</li>
        <li><b>IDMapper</b> displays a report of how many identifiers were mapped. Make note of this information as it impacts all down-stream analysis; If the mapping was unsuccessful, downstream analysis will be as well.</li>
        <li>Next, we can load the data. Download a local copy of <a href="https://cytoscape.github.io/cytoscape-tutorials/protocols/data/TCGA-Ovarian-MesenvsImmuno_data.csv">TCGA-Ovarian-MesenvsImmuno_data.csv</a>. Then go to <highlight>File &rarr; Import &rarr; Table from File...</highlight> and select the data file.</li>
    </ul>
</section>
 
<section>
    <h1>Data integration</h1>
    <p>In the <highlight>Import Columns from Table</highlight> interface, there are a few things that need to be changed from the defaults:</p>
            <ul>
                <li><highlight>Key Column for Network</highlight> should be <highlight>Entrez Gene</highlight>.</li>
                <li>In the data preview, <highlight>Gene ID</highlight> should be the key, if not you can set it by right-clicking on the column header and selecting the key symbol.</li>
                <li>The <highlight>Import Columns from Table</highlight> interface should now look like this:</li>
            </ul><br /><br />
<img class="plain" src="TableImport.png" style="height:500px">    
</section>  
                
<section>
    <h1>Visualization</h1>
    <p>To complete the import, click <highlight>OK</highlight>. You will notice two new columns in the <highlight>Node Table</highlight>. We can now use this data to create a network visualization.</p>   
    <ul>
        <li>For node <highlight>Fill Color</highlight>, first delete the existing mapping and then create a continuous mapping for <highlight>logFC</highlight>, with the default blue-red gradient.</li>
        <li>Change the default node <highlight>Fill Color</highlight> to light grey.</li>
	</ul>
	        <div class="fragment">
                <p><i>Pro-tip: If you apply the <highlight>yFiles Organic</highlight> layout, you can end up with network views like this...</i></p>
	    <img class="plain" src="string-ovarian-styled.png" style="height:520px">   
	    </div>
</section>
                
                
<section>
    <h1>Visualization</h1>
    <p>The TCGA found several genes that were commonly mutated in ovarian cancer, so called "cancer drivers". We can add information about these genes to the network visualization, by changing the visual style of these nodes. Three of the most important drivers are <b>TP53</b>, <b>BRCA1</b> and <b>BRCA2</b>. We will add a thicker, clored border for these genes in the network.</p>
    <ul>
        <li>Select all three driver genes by holding down the Command key while clicking each of the genes.</li>
        <li>In the <highlight>Style</highlight> panel, add a style bypass for node <highlight>Border Width</highlight> (5) and node <highlight>Border Paint</highlight> (bright pink). You can create a style bypass by clicking the <highlight>Bypass</highlight> (<b>Byp.</b>) column for each attribute.</li>
    </ul>
</section>
    
            
<section>
<p>The network will now look like this:</p>
<img class="plain" src="string-ovarian-drivers.png" style="height:750px">    
</section>
                
<section>
    <h1>Other Analysis Options</h1>
    <ul>
        <li>Exploring networks: finding paths, hubs and modules (clusterMaker, MCODE, jActiveModules, NetworkAnalyzer, PathFinder)</li>
        <li>Extending networks with Transcription Factors, miRNAs, etc using CyTargetLinker</li>
    
    </ul>
</section>
                
<section>
    <h1>Exporting Networks</h1>
    <p>Cytoscape provides a number of ways to export results and visualizations:</p>
	<ul>
        <li>As an image: <highlight>File &rarr; Export &rarr; Network to Image...</highlight></li>
        <li>To a public repository: <highlight>File &rarr; Export &rarr; Network to NDEx</highlight>, or <highlight>File &rarr; Export &rarr; Collection to NDEx</highlight></li>
		<li>As a Cytoscape JSON file: <highlight>File &rarr; Export &rarr; Network to File</highlight> and select <b>Cytoscape.js JSON</b> as the format.</li>
	</ul>
</section>
                
<!-- THIS IS WHERE YOUR CONTENT ENDS -->
<!-- ========================================================== -->
            
            </div>
		</div>

		<script src="../../lib/js/head.min.js"></script>
		<script src="../../js/reveal.js"></script>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-155159-15', 'auto');
  ga('send', 'pageview');

</script>
<script>
	Reveal.addEventListener( 'ready', function() {
	if(!window.navigator.userAgent.includes('CyBrowser')){
		var divs = document.getElementsByClassName("cybrowser")
		for (var i=0;i<divs.length;i++){
			divs[i].style.display = "none";
		}
	} else {
		var divs = document.getElementsByClassName("not-cybrowser")
		for (var i=0;i<divs.length;i++){
			divs[i].style.display = "none";
		}
	}
});
</script>  
		<script>
			// More info https://github.com/hakimel/reveal.js#configuration
			Reveal.initialize({
				history: true,
				controls: true,
				progress: true,
				slideNumber: false,
				touch: true,
				fragments: true,
				help: true,
				showNotes: false,
				previewLinks: false,
				transition: 'slide', // none/fade/slide/convex/concave/zoom
				width: 1200,
				height: 860,
				autoSlide: 0,

				// More info https://github.com/hakimel/reveal.js#dependencies
				dependencies: [
				{ src: '../../plugin/markdown/marked.js' },
				{ src: '../../plugin/markdown/markdown.js' },
				{ src: '../../plugin/markdown/zoom.js', async: true, condition: function(){ return !!document.body.classList; } },
				{ src: '../../plugin/zoom-js/zoom.js', async: true, condition: function(){ return !!document.body.classList; } },
				{ src: '../../plugin/notes/notes.js', async: true },
				{ src: '../../plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
				{ src: '../../plugin/external/external.js', condition: function() { return !!document.querySelector( '[data-external]' ); } }
				]
				});
			</script>
		</body>
	</html>
